tensorflow/core/kernels/cudnn_rnn_ops.cc

```diff
@@ -650,20 +687,30 @@ Status DoForward(OpKernelContext* context, const RnnDescriptor& rnn_desc,
                  Tensor* output, Tensor* output_h, Tensor* output_c,
                  ScratchAllocator* reserve_space_allocator,
                  ScratchAllocator* workspace_allocator,
-                 ProfileResult* output_profile_result) {
+                 ProfileResult* output_profile_result,
+                 const Tensor* sequence_lengths=nullptr) {
   std::unique_ptr<RnnSequenceTensorDescriptor> input_desc;
   std::unique_ptr<RnnStateTensorDescriptor> state_desc;
   std::unique_ptr<RnnSequenceTensorDescriptor> output_desc;
 
+  absl::Span<const int> seq_lengths;
+  if (sequence_lengths != nullptr) {
+    DeviceMemory<int> sequence_lengths_data = 
```

Comment:
No need to convert to `DeviceMemory`. Use something like `tensor->template flat<T>().data()`.

Same for other places.